stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModelBinding.kt

```diff
@@ -26,8 +26,19 @@ public fun MessageListViewModel.bindView(view: MessageListView, lifecycleOwner:
     view.setOnMessageRetryHandler { onEvent(RetryMessage(it)) }
 
     state.observe(lifecycleOwner) { state ->
-        if (state is MessageListViewModel.State.Result) {
-            view.displayNewMessage(state.messageListItem)
+        when (state) {
+            is MessageListViewModel.State.Loading -> {
+                view.showLoadingView()
+            }
```

Comment:
We should hide the empty state view on this branch. Even if right now we never get into a `Loading` state after `Result`, it's a cleaner implementation of binding state to UI if we get the binding right from any previous state.